src/lib.rs

```diff
@@ -776,6 +779,88 @@ impl Client {
         })
     }
 
+    /// Gets the current window handle.
+    pub async fn get_window_handle(
+        &mut self,
+    ) -> Result<webdriver::common::WebWindow, error::CmdError> {
+        let res = self.issue(WebDriverCommand::GetWindowHandle).await?;
+        match res {
+            Json::String(x) => Ok(webdriver::common::WebWindow(x)),
+            v => Err(error::CmdError::NotW3C(v)),
+        }
+    }
+
+    /// Gets a list of all active windows (and tabs)
+    pub async fn get_window_handles(
+        &mut self,
+    ) -> Result<Vec<webdriver::common::WebWindow>, error::CmdError> {
+        let res = self.issue(WebDriverCommand::GetWindowHandles).await?;
+        match res {
+            Json::Array(handles) => handles
+                .into_iter()
+                .map(|handle| match handle {
+                    Json::String(x) => Ok(webdriver::common::WebWindow(x)),
+                    v => Err(error::CmdError::NotW3C(v)),
+                })
+                .collect::<Result<Vec<_>, _>>(),
+            v => Err(error::CmdError::NotW3C(v)),
+        }
+    }
+
+    /// Switches to the chosen window.
+    pub async fn switch_to_window(
+        &mut self,
+        window: webdriver::common::WebWindow,
+    ) -> Result<(), error::CmdError> {
+        let params = SwitchToWindowParameters { handle: window.0 };
+        let _res = self.issue(WebDriverCommand::SwitchToWindow(params)).await?;
+        Ok(())
+    }
+
+    /// May close the session if no other windows exist.
```

Comment:
This could use some more documentation. What happens if other windows _do_ exist? Is this usually used to close _another_ window, and if so, should we provide a method that takes a `WebWindow` and closes it?